x-pack/plugins/security_solution/public/common/components/toolbar/bulk_actions/alert_bulk_tags.tsx

```diff
@@ -0,0 +1,178 @@
+        addWarning({
+          title: 'Warning',
+          text: `${updated} alerts updated successfully, but ${conflicts} didn't due to version conflicts`,
+        });
+      } else {
+        addSuccess(`${updated} alerts successfully updated`);
+      }
+    },
+    [addSuccess, addWarning]
+  );
+
+  const onUpdateFailure = useCallback(
+    (error: Error) => {
+      addError(error.message, { title: 'Tags failed to update' });
+    },
+    [addError]
+  );
+
+  const [selectableAlertTags, setSelectableAlertTags] =
+    useState<EuiSelectableOption[]>(initalTagsState);
+
+  const onTagsUpdate = useCallback(async () => {
+    closePopoverMenu();
+    const ids = alertIds.map((item) => item._id);
+    const query: Record<string, unknown> = getUpdateAlertsQuery(ids).query;
+    const tagsToAddArray = Object.keys(tagsToAdd);
+    const tagsToRemoveArray = Object.keys(tagsToRemove);
+    try {
+      setIsLoading(true);
+
+      const response = await setAlertTags({
+        tags: { tags_to_add: tagsToAddArray, tags_to_remove: tagsToRemoveArray },
+        query,
+      });
+
+      setIsLoading(false);
+      refetchQuery();
+      if (refresh) refresh();
+      if (clearSelection) clearSelection();
+
+      if (response.version_conflicts && ids.length === 1) {
+        throw new Error('Updated failed due to version conflicts');
+      }
+
+      onUpdateSuccess(response.updated ?? 0, response.version_conflicts ?? 0);
+    } catch (err) {
+      onUpdateFailure(err);
+    }
+  }, [
+    closePopoverMenu,
+    alertIds,
+    tagsToAdd,
+    tagsToRemove,
+    setIsLoading,
+    setAlertTags,
+    refetchQuery,
+    refresh,
+    clearSelection,
+    onUpdateSuccess,
+    onUpdateFailure,
+  ]);
+
+  const handleTagsOnChange = (
+    newOptions: EuiSelectableOption[],
+    event: EuiSelectableOnChangeEvent,
+    changedOption: EuiSelectableOption
+  ) => {
+    if (changedOption.checked === 'off') {
+      // Don't allow intermediate state when selecting, only from initial state
+      newOptions[newOptions.findIndex((option) => option.label === changedOption.label)] = {
+        ...changedOption,
+        checked: undefined,
+      };
+      tagsToRemove[changedOption.label] = true;
+      delete tagsToAdd[changedOption.label];
+    } else if (changedOption.checked === 'on') {
+      tagsToAdd[changedOption.label] = true;
+      delete tagsToRemove[changedOption.label];
+    } else if (!changedOption.checked) {
+      tagsToRemove[changedOption.label] = true;
+      delete tagsToAdd[changedOption.label];
+    }
+    setSelectableAlertTags(newOptions);
+  };
+
+  return (
+    <>
+      <EuiSelectable
+        allowExclusions
+        searchable
+        searchProps={{
+          placeholder: 'Search tags',
+        }}
+        aria-label={'search them tags'}
+        options={selectableAlertTags}
+        onChange={handleTagsOnChange}
+        emptyMessage={'im empty fill me up'}
+        noMatchesMessage={'no matches here you absolute fool'}
```

Comment:
🤣 just pointing out we'll need to update the text so we don't forget and push with this